gdx/src/com/badlogic/gdx/maps/tiled/TmxMapLoader.java

```diff
@@ -83,6 +83,51 @@ public TiledMap load (String fileName, TmxMapLoader.Parameters parameter) {
 		return map;
 	}
 
+	/** Loads a tile set from the given file. The file is resolved via the {@link FileHandleResolver} set in the constructor of
+	 * this class. By default it will resolve to an internal file.
+	 *
+	 * @param fileName the filename of the tile set
+	 * @param map the TiledMap to which the tile set will be added
+	 * @return the TiledMap with the loaded tile set */
+	public TiledMap loadCustomTileSet (String fileName, TiledMap map) {
+		return loadCustomTileSet(fileName, map, new TmxMapLoader.Parameters());
+	}
+
+	/** Loads a tile set from the given file. The file is resolved via the {@link FileHandleResolver} set in the constructor of
+	 * this class. By default it will resolve to an internal file.
+	 *
+	 * @param fileName the filename of the tile set
+	 * @return the TiledMap with the loaded tile set */
+	public TiledMap loadCustomTileSet (String fileName) {
+		TiledMap map = new TiledMap();
+		return loadCustomTileSet(fileName, new TiledMap(), new TmxMapLoader.Parameters());
+	}
+
+	/** Loads a tile set from the given file. The file is resolved via the {@link FileHandleResolver} set in the constructor of
+	 * this class. By default it will resolve to an internal file.
+	 *
+	 * @param fileName the filename of the tile set
+	 * @param map the TiledMap to which the tile set will be added
+	 * @param parameter specifies whether to use y-up, generate mip maps etc.
+	 * @return the TiledMap with the loaded tile set */
+	public TiledMap loadCustomTileSet (String fileName, TiledMap map, TmxMapLoader.Parameters parameter) {
+		FileHandle tmjFile = resolve(fileName);
+		Element tileSet = xml.parse(tmjFile);
```

Comment:
I'm not at all familiar with Tiled and its format, but `xml.parse(tmjFile);` looks suspicious. TMJ is JSON-based, so why would it be parsable with an XML reader?

Reply:
The change is in the TmxMapLoader, so parsing XML should be correct. Only the name of the FileHandle is misleading

Reply:
Would you fix this? @Kevin-Garnier

Reply:
Yes sorry, I was busy with work, I'll add all your remarks soon